quickwit/quickwit-ui/src/components/QueryEditor/AggregationEditor.tsx

```diff
@@ -0,0 +1,305 @@
+    const updatedSearchRequest = Object.assign({}, props.searchRequest, {aggregationConfig: updatedAggregation});
+    props.onSearchRequestUpdate(updatedSearchRequest);
+  };
+
+  const handleAggregationChange = (pos: number, event: SelectChangeEvent) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      switch(value) {
+        case "histogram": {
+          newAggregations[pos] = {
+            histogram: {
+              interval: "auto",
+            }
+          };
+          break;
+        }
+        case "term": {
+          newAggregations[pos] = {
+            term: {
+              field: "",
+              size: 10,
+            }
+          };
+          break;
+        }
+        case "rm": {
+          console.log("av", newAggregations);
+          newAggregations.splice(pos, 1);
+          console.log("cesar", newAggregations);
+        }
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  const handleHistogramChange = (pos: number, event: SelectChangeEvent) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      newAggregations[pos] = {histogram: {interval:value}};
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  }
+
+  const handleTermFieldChange = (pos: number, event: React.ChangeEvent<HTMLInputElement|HTMLTextAreaElement>) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      const term = newAggregations[pos]
+      if (isTerm(term)) {
+        term.term.field = value;
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  const handleTermCountChange = (pos: number, event: React.ChangeEvent<HTMLInputElement|HTMLTextAreaElement>) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      const term = newAggregations[pos]
+      if (isTerm(term)) {
+        term.term.size = Number(value);
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  function isHistogram(agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined): agg is {histogram: HistogramAgg} {
+    if (!agg) return false;
+    return Object.hasOwn(agg, "histogram")
+  }
+
+  function isTerm(agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined): agg is {term: TermAgg} {
+    if (!agg) return false;
+    return Object.hasOwn(agg, "term")
+  }
+
+  const getAggregationKind = (agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined) => {
+    if (isHistogram(agg)) {
+      return "histogram";
+    }
+    if (isTerm(agg)) {
+      return "term";
+    }
+    return "new";
+  };
+
+  const makeOptions = (pos: number, agg: ({term: TermAgg} | {histogram: HistogramAgg})[]) => {
```

Comment:
Returning an array of elements requires each element to have a `key` prop or react will complain.
It is usually more practical to return the elements wrapped within a fragment `<>…</>.

Reply:
in this case, it seemed easier to use key (no need for code change, and each element can have a reasonable key in form of its value)